linode/volume/framework_model.go

```diff
@@ -0,0 +1,64 @@
+package volume
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+)
+
+type VolumeModel struct {
+	ID             types.Int64  `tfsdk:"id"`
+	Label          types.String `tfsdk:"label"`
+	Region         types.String `tfsdk:"region"`
+	Size           types.Int64  `tfsdk:"size"`
+	LinodeID       types.Int64  `tfsdk:"linode_id"`
+	FilesystemPath types.String `tfsdk:"filesystem_path"`
+	Tags           types.Set    `tfsdk:"tags"`
+	Status         types.String `tfsdk:"status"`
+	Created        types.String `tfsdk:"created"`
+	Updated        types.String `tfsdk:"updated"`
+}
+
+func (data *VolumeModel) parseComputedAttributes(
+	ctx context.Context,
+	volume *linodego.Volume,
+) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	data.ID = types.Int64Value(int64(volume.ID))
+	data.Status = types.StringValue(string(volume.Status))
+	data.Region = types.StringValue(volume.Region)
+	data.Size = types.Int64Value(int64(volume.Size))
+
+	if volume.LinodeID != nil {
+		data.LinodeID = types.Int64Value(int64(*volume.LinodeID))
+	} else {
+		data.LinodeID = types.Int64Null()
+	}
```

Comment:
Just a heads up that this is _technically_ a breaking change:
```terraform
data "linode_volume" "test" {
  id = 1281734
}

output "is_detached" {
  value = data.linode_volume.test.linode_id == 0 // evaluates to false for detached linodes
}
```

Reply:
Breaking changes removed, thanks!